### PR TITLE
libplacebo: use upstreamed patch

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

Fetch patch from upstream repository. 
See #108727

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by baloo** [comment]
cc  @primeos

File: pkgs/development/libraries/libplacebo/default.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/libplacebo/default.nix:22]:
```suggestion
      # support glslang>=11.0.0; Upstream MR: https://code.videolan.org/videolan/libplacebo/-/merge_requests/131
```

**Review by primeos — CHANGES_REQUESTED** [review]
I don't see the advantage here tbh. Currently the commit is not yet merged into the upstream repository and that commit is likely GCed when/if you delete your branch (for the upstream MR).

In case upstream requests any changes (e.g. to make it work with older glslang versions as well) we could switch to the newer upstreamed patch but else this only documents it a bit better until the next release (but I guess we can also merge it for that reason if you want - should wait until the MR is merged though).

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>plex-media-player: log was empty</li>
  </ul>
</details>
<details>
<summary>
18 packages built:
</summary>
<ul>
<li>celluloid</li>
<li>curseradio</li>
<li>jellyfin-mpv-shim</li>
<li>jftui</li>
<li>libplacebo</li>
<li>minitube</li>
<li>mpc-qt</li>
<li>mpv</li>
<li>mpv-unwrapped</li>
<li>mpvScripts.mpris</li>
<li>plex-mpv-shim</li>
<li>python37Packages.mpv</li>
<li>python38Packages.mpv</li>
<li>python39Packages.mpv</li>
<li>qimgv</li>
<li>somafm-cli</li>
<li>sublime-music</li>
<li>tuijam</li>
</ul>
</details>

**Comment by baloo** [comment]
> I don't see the advantage here tbh. Currently the commit is not yet merged into the upstream repository and that commit is likely GCed when/if you delete your branch (for the upstream MR).
> 
> In case upstream requests any changes (e.g. to make it work with older glslang versions as well) we could switch to the newer upstreamed patch but else this only documents it a bit better until the next release (but I guess we can also merge it for that reason if you want - should wait until the MR is merged though).

I understood the policy was to upstream patches and not have them live in nixpkgs tree as much as possible. But I could be wrong :)

**Comment by baloo** [comment]
(feel free to close if I'm incorrect about the policy. No hard feeling)

**Comment by primeos** [comment]
> I understood the policy was to upstream patches and not have them live in nixpkgs tree as much as possible. But I could be wrong :)

Yes, we try to upstream the patches to drop them after the next release / try to keep our downstream patches to a minimum. But if they're already in Nixpkgs I don't think there's any reason/recommendation to switch to fetchurl (that only makes sense if the patch was upstreamed in advance to link to the source and make the patch easier to verify; and for large patches to avoid bloating Nixpkgs). But tbh this is just how I view/perceive it so I could be wrong (but I hope it makes sense anyway).

But since it seems like upstream might indeed want the patch to be backwards compatible we could switch to it once the MR is merged (e.g. for the unlikely case the the gslang update gets reverted in Nixpkgs). Feel free to update this PR or ping me once the MR is merged or to close this PR.

**Comment by baloo** [comment]
sounds good

**Comment by SuperSandro2000** [comment]
> I don't see the advantage here tbh.

It is the preferred way because we do not need to maintain the patch here.

> Currently the commit is not yet merged into the upstream repository and that commit is likely GCed when/if you delete your branch (for the upstream MR).

That is not an issue. The patch is cached when it is merged and if it is merged we can always update the commit and the hash which is way easier than maintaining the patch in tree.

> I understood the policy was to upstream patches and not have them live in nixpkgs tree as much as possible. But I could be wrong :)

Yes, it is.

**Comment by primeos** [comment]
> It is the preferred way because we do not need to maintain the patch here.

@SuperSandro2000 please explain... We don't need to maintain our downstream patch, we can/must drop if after the next release anyway.

> That is not an issue. The patch is cached when it is merged

No. It might be cached once it is first build on Hydra but not immediately when merging. And I'm not sure if tarballs.nixos.org even caches patches as well (would make sense but I didn't check it).

> and if it is merged we can always update the commit and the hash which is way easier than maintaining the patch in tree.

We're not maintaining the patch here, it is already upstreamed (that is the important thing, not where we source the patch from).

> Yes, it is.

Agreed. Maybe your first statements came of wrong then but in any case: This PR is basically a no-op. But since the MR is merged now and we already have this PR (I usually try not to reject PRs :o) we can just merge it and be done with it.

@GrahamcOfBorg build libplacebo

@baloo Anyway, sorry that this discussion went a bit in other directions, and thanks again for the quick fix and upstreaming the patch! :)

**Pull request merged**
